### PR TITLE
fix(i18n): compatible with timestamp in milliseconds or null values

### PR DESCRIPTION
**Pull request by Justineo**
# Summary

The filter API is returning timestamps in milliseconds instead of seconds and `null` as `updated_at` values for those weren't modified since created. The issue affects both filtered services and routes.

Before:

image

After:

image

### CODE REVIEW

File: packages/core/i18n/src/i18n.ts [hunk 1]
```diff
@@ -52,19 +52,32 @@ export const createI18n = <MessageSource extends Record<string, any>>
...
    * Formats a unix timestamp into a formatted date string
    * @param {Number} timestamp a unix timestamp in seconds
    * @returns a date string formatted like 'Apr 6, 2022 10:50'
    */
-  const formatUnixTimeStamp = (timestamp: number): string => {
+  const formatUnixTimeStamp = (timestamp?: number | null): string => {
     const invalidDate = 'Invalid Date'
     if (!timestamp) {
-      return invalidDate
+      return '-'
```
Comment (ValeryG) [inline packages/core/i18n/src/i18n.ts:76]: not saying this is wrong change, but would some places where user expects to see `Invalid Date` and now sees `-` be "broken" or "changed without the user's content" ? Also are we sure there are no tests that expects `Invalid Date` ? 
Seems that the change is not related to shameful normalize and can be excluded from this PR, unless it's absolutely needed?

Comment (Justineo) [reply packages/core/i18n/src/i18n.ts:76]: I'm not sure if there are valid use cases where users should see `Invalid Date`. I can revert this change, as it was intended to handle potential `null` values returned by backend APIs, and we are already providing the correct fallback value (use `created_at` if `updated_at` is `null`). I’m just curious to understand if there are any scenarios in our current design where we might expect to see `Invalid Date`.

Comment (ValeryG) [reply packages/core/i18n/src/i18n.ts:76]: I don't see this string anywhere in the tests. So looks like safe change. If someone complains we just file JIRA and fix :)

Comment (Justineo) [reply packages/core/i18n/src/i18n.ts:76]: Sounds good. Can you stamp on this one?

Comment (kaiarrowood) [inline packages/core/i18n/src/i18n.ts:76]:
```suggestion
      return '—'
```

Can we use emdash instead please?

Comment (Justineo) [reply packages/core/i18n/src/i18n.ts:76]: I think I should just remove this. We are already handling `null` before this.

File: packages/core/i18n/src/i18n.ts [hunk 2]
```diff
@@ -52,6 +52,19 @@ export const createI18n = <MessageSource extends Record<string, any>>
   const { $t, ...otherProps } = intlOriginal
   const intl = otherProps
 
+  /**
+   * Shamefully normalize a timestamp to be in seconds as some APIs are returning timestamps in milliseconds
+   * TODO: Remove this function once all timestamps are normalized from the backend
```
Comment (Justineo) [inline packages/core/i18n/src/i18n.ts:57]: Ticket for the backend: [MA-3412](https://konghq.atlassian.net/browse/MA-3412)

**Comment by Justineo** [comment]
I've updated the code and no longer handles `null` value when formatting the timestamp. PTAL @ValeryG @kaiarrowood

**Review by kaiarrowood — APPROVED** [review]
LGTM! 🚀
